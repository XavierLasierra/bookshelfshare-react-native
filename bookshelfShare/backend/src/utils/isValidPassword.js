const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
