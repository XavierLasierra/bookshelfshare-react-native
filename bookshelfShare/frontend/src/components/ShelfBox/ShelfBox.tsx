import React from 'react';
import { TouchableOpacity, View } from 'react-native';

import styles from './shelfBox.styles';

export default function ShelfBox({
  width, clickCallback, row, column, activeShelf
}: any) {
  return (
    <View style={[styles.shelfBoxOutline, { width }]}>
      <TouchableOpacity
        style={[
          styles.shelfInside,
          JSON.stringify(activeShelf) === JSON.stringify([row, column]) && styles.activeShelf]}
        onPress={() => clickCallback([row, column])}
      />
    </View>
  );
}
