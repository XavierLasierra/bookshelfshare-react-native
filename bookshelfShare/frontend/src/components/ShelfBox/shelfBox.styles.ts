import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  shelfBoxOutline: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: stylesConstants.colors.shelfOutline,
    height: '100%'
  },
  shelfInside: {
    width: '90%',
    height: '90%',
    backgroundColor: stylesConstants.colors.shelfBackground,
    borderColor: stylesConstants.colors.shelfShadow,
    borderBottomWidth: 5,
    borderRightWidth: 5
  }
});
