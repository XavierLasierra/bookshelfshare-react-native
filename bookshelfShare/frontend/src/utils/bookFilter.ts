export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
