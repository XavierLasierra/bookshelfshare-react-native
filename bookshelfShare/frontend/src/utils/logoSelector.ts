import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
