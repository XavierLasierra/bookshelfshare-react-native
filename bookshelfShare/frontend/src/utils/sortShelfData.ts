export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
