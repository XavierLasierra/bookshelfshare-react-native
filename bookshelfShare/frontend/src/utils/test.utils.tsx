import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
