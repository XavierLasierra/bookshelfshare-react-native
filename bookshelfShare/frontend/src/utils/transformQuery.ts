interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
