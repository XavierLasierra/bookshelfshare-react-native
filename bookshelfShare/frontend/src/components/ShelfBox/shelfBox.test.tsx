import React from 'react';
import ShelfBox from './ShelfBox';
import {render, fireEvent} from '../../utils/test.utils';

describe('Given a ShelfBox component', () => {
  describe('When it is rendered', () => {
    describe('And activeShelf is different from [row,column]', () => {
      describe('And numberOfBooks is 1', () => {
        test('Then should match the snapshot', () => {
          const clickCallback = jest.fn();
          const screen = render(
            <ShelfBox
              width={100}
              clickCallback={clickCallback}
              row={0}
              column={0}
              numberOfBooks={1}
              activeShelf={[1, 1]}
            />,
          );
          expect(screen).toMatchSnapshot();
        });
      });
      describe('And numberOfBooks 2', () => {
        test('Then should match the snapshot', () => {
          const clickCallback = jest.fn();
          const screen = render(
            <ShelfBox
              width={100}
              clickCallback={clickCallback}
              row={0}
              column={0}
              numberOfBooks={2}
              activeShelf={[1, 1]}
            />,
          );
          expect(screen).toMatchSnapshot();
        });
      });
    });
    describe('And activeShelf is equal to [row,column]', () => {
      let clickCallback: any;
      let screen: any;
      beforeEach(() => {
        clickCallback = jest.fn();
        screen = render(
          <ShelfBox
            width={100}
            clickCallback={clickCallback}
            row={0}
            column={0}
            numberOfBooks={1}
            activeShelf={[0, 0]}
          />,
        );
      });
      test('Then should match the snapshot', () => {
        expect(screen).toMatchSnapshot();
      });

      describe('And the rendered element is pressed', () => {
        test('Then clickCallback should be called with [row, column]', () => {
          const currentBoxButton = screen.getByTestId('shelf-0-0');
          fireEvent.press(currentBoxButton);

          expect(clickCallback).toHaveBeenCalledWith([0, 0]);
        });
      });
    });
  });
});
