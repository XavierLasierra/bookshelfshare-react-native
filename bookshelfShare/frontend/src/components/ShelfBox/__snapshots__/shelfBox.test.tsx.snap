// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ShelfBox component When it is rendered And activeShelf is different from [row,column] And numberOfBooks 2 Then should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#958472",
        "height": "100%",
        "justifyContent": "center",
      },
      Object {
        "width": 100,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4D3C2A",
        "borderBottomWidth": 5,
        "borderColor": "#2F2406",
        "borderRightWidth": 5,
        "height": "90%",
        "justifyContent": "center",
        "opacity": 1,
        "width": "90%",
      }
    }
    testID="shelf-0-0"
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      2
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      books
    </Text>
  </View>
</View>
`;

exports[`Given a ShelfBox component When it is rendered And activeShelf is different from [row,column] And numberOfBooks is 1 Then should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#958472",
        "height": "100%",
        "justifyContent": "center",
      },
      Object {
        "width": 100,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4D3C2A",
        "borderBottomWidth": 5,
        "borderColor": "#2F2406",
        "borderRightWidth": 5,
        "height": "90%",
        "justifyContent": "center",
        "opacity": 1,
        "width": "90%",
      }
    }
    testID="shelf-0-0"
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      1
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      book
    </Text>
  </View>
</View>
`;

exports[`Given a ShelfBox component When it is rendered And activeShelf is equal to [row,column] Then should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#958472",
        "height": "100%",
        "justifyContent": "center",
      },
      Object {
        "width": 100,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#BF7B54",
        "borderBottomWidth": 5,
        "borderColor": "#2F2406",
        "borderRightWidth": 5,
        "height": "90%",
        "justifyContent": "center",
        "opacity": 1,
        "width": "90%",
      }
    }
    testID="shelf-0-0"
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      1
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      book
    </Text>
  </View>
</View>
`;
